cms/djangoapps/contentstore/rest_api/v1/views.py

```diff
@@ -58,45 +62,106 @@ def proctored_exam_settings(request, course_id):
+    POST
+    ------------------------------------------------------------------------------------
+
+    **Returns**
+
+        * 200: OK - Contains a set of course proctored exam settings.
+        * 400: Bad Request - Unable to save requested settings.
+        * 401: The requesting user is not authenticated.
+        * 403: The requesting user lacks access to the course.
+        * 404: The requested course does not exist.
+
+    **Response**
+
+        In the case of a 200 response code, the response will echo the updated proctored
+        exam settings data.
     """
-    course_key = CourseKey.from_string(course_id)
-    with modulestore().bulk_operations(course_key):
-        if request.method == 'GET':
-            course_module = get_course_and_check_access(course_key, request.user)
+    PROCTORED_EXAM_SETTINGS_KEYS = [
+        'enable_proctored_exams',
+        'allow_proctoring_opt_out',
+        'proctoring_provider',
+        'proctoring_escalation_email',
+        'create_zendesk_tickets',
+    ]
+
+    def get(self, request, course_id):
+        """ GET handler """
+        course_module = self._get_and_validate_course(request.user, course_id)
+        course_metadata = CourseMetadata().fetch_all(course_module)
+        proctored_exam_settings = self._get_proctored_exam_setting_values(course_metadata)
+
+        data = {}
+
+        data['proctored_exam_settings'] = proctored_exam_settings
+        data['available_proctoring_providers'] = get_available_providers()
+        data['course_start_date'] = course_metadata['start'].get('value')
+        data['is_staff'] = request.user.is_staff
+
+        serializer = ProctoredExamConfigurationSerializer(data)
+
+        return Response(serializer.data)
+
+    def post(self, request, course_id):
+        """ POST handler """
+        exam_config = ProctoredExamSettingsSerializer(data=request.data.get('proctored_exam_settings', {}))
+        exam_config.is_valid(raise_exception=True)
+        with modulestore().bulk_operations(CourseKey.from_string(course_id)):
+            course_module = self._get_and_validate_course(request.user, course_id)
+            course_metadata = CourseMetadata().fetch_all(course_module)
 
-            if not course_module:
+            models_to_update = {}
+            for setting_key, value in exam_config.data.items():
+                model = course_metadata.get(setting_key)
+                if model:
+                    models_to_update[setting_key] = copy.deepcopy(model)
+                    models_to_update[setting_key]['value'] = value
+
+            # validate data formats and update the course module object
+            is_valid, errors, updated_data = CourseMetadata.validate_and_update_from_json(
+                course_module,
+                models_to_update,
+                user=request.user,
+            )
+
+            if not is_valid:
+                error_messages = [error.get('message') for error in errors]
```

Comment:
I think it would be valuable to tie the error message to the field that the error message is for. Otherwise, from the client, it won't be possible to know programmatically which field is incorrect, given the error.

For example, 
```
{ "proctoring_provider": "The proctoring provider cannot be modified after a course has started. Contact support@example.com for assistance." }
```

Reply:
humm I'll have to look into this.  CourseMetadata just returns the errors by not the key for the model they are attributed to

Reply:
If it's not possible, we can find a way around it. It would be useful to have this information so that we can link to the field on the page that has the error, which is necessary for greater accessibility.

Reply:
turns out this was easy to add w/o breaking anything